src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexAhoCorasick.cs

```diff
@@ -0,0 +1,197 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+// The RegexAhoCorasick object once ready, runs the Aho-Corasick algorithm to find multiple string matches in the pattern efficiently.
+
+using System.Collections.Generic;
+
+namespace System.Text.RegularExpressions
+{
+    internal class TrieNode
```

Comment:
struct?

Reply:
Yup, at some point. Right now it'd just force me to use annoying code patterns